test/src/unit-filter-pipeline.cc

```diff
@@ -4087,25 +4087,15 @@ void testing_xor_filter() {
   tile.init_unfiltered(
       constants::format_version, t, tile_size, cell_size, dim_num);
 
-  // The MSVC Windows compiler doesn't support std::uniform_int_distribution
-  // with
-  // type int8_t instantiation, so we use this little hack to generate the right
-  // type for our int distribution.
-  std::random_device rd;
-  std::mt19937 gen(rd());
-  typedef typename std::conditional<
-      is_windows && std::is_same<T, int8_t>::value,
-      int16_t,
-      T>::type DIST_TYPE;
-
   // Setting up the random number generator for the XOR filter testing.
-  std::uniform_int_distribution<DIST_TYPE> dis(
+  std::mt19937 gen(0x57A672DE);
```

Comment:
Since you're using 64-bit integers below, you should be using `std::mt19937_64`. What you've got here is a 32-bit generator.
https://en.cppreference.com/w/cpp/numeric/random/mersenne_twister_engine